m2m-transfers/V3/REPWRITERSPECS/BANNO.M2MTRANSFERS.V3.POW

```diff
@@ -659,103 +680,184 @@ END [PRINT]
+      END
+
+     IF VALIDATESTATUS=TARGETVERIFIED AND
+        UPPERCASE(TARGETACCTTYPE)="SHARE" AND
+        TARGETSLID="" THEN
+      DO
+       FINDSHARECODE=1
+       CALL FINDBYSHARECODE
+       IF SLFOUND=FALSE THEN
         DO
+         FINDSHARECODE=0
+         CALL FINDBYSHARECODE
+        END
 
-         FOR EACH LOAN WITH (LOAN:ID=TARGETSLID AND
-                             LOAN:CLOSEDATE=DATENULL AND
-                             LOAN:CHARGEOFFDATE=DATENULL AND
-                             LOAN:PAYOFFAMOUNT>$0.00)
+       IF SLFOUND=FALSE THEN
+        VALIDATESTATUS=TARGETNOTFOUNDBYTYPE
+      END    
+   
+     IF VALIDATESTATUS=TARGETVERIFIED AND
+        UPPERCASE(TARGETACCTTYPE)="LOAN" THEN
+      DO
+       SLFOUND=FALSE
+       FOR EACH LOAN WITH (LOAN:ID=TARGETSLID)
+        DO
+         SLFOUND=TRUE
+         IF LOAN:CLOSEDATE<>DATENULL OR LOAN:CHARGEOFFDATE<>DATENULL THEN
+          DO
+           VALIDATESTATUS=TARGETSLCLOSED
+           IF LOAN:CLOSEDATE<>DATENULL THEN
+            TARGETSLCLOSECODATE=LOAN:CLOSEDATE
+           ELSE IF LOAN:CHARGEOFFDATE<>DATENULL THEN
+            TARGETSLCLOSECODATE=LOAN:CHARGEOFFDATE 
+          END
+         ELSE IF LOAN:PAYOFFAMOUNT<=$0.00 THEN
+          VALIDATESTATUS=TARGETLOANZEROPAYOFF
+  
+         IF VALIDATESTATUS=TARGETVERIFIED THEN
           DO
            SVCCODEFOUND=FALSE
-           FOR TMPLOOP=1 TO SVCCODEMAX
+           TMPLOOP=1
+           WHILE TMPLOOP<=SVCCODEMAX AND SVCCODEFOUND=FALSE
             DO
-             IF SERVICEIN(TMPLOOP)=TRUE AND
-                ANYSERVICE(LOAN,TMPLOOP) THEN
-              DO
-               SVCCODEFOUND=TRUE
-               SLFOUND=TRUE
-               TMPLOOP=SVCCODEMAX
-              END
+             IF SERVICEIN(TMPLOOP)=TRUE AND ANYSERVICE(LOAN,TMPLOOP) THEN
+              SVCCODEFOUND=TRUE
+        
+             TMPLOOP=TMPLOOP+1
             END
-           IF SVCCODEFOUND=TRUE THEN
+           IF SVCCODEFOUND=FALSE THEN
+            VALIDATESTATUS=TARGETNOSERVICECODE
+           ELSE
             DO
              BNODEBUGMSG="Loan ID found - BY TARGET TYPE: "+LOAN:ID
              CALL BNODEBUGMSGADD
             END
-          END UNTIL SLFOUND=TRUE
+          END  
         END
+       UNTIL SLFOUND=TRUE 
+    
+       IF SLFOUND=FALSE THEN
+        VALIDATESTATUS=TARGETNOTFOUNDBYSLID
       END
-    END [IF TARGETACCTCLOSEDATE=DATENULL]
-  END [FOR ACCOUNT]
+    END [FOR ACCOUNT]
+  END  
 
-
- BNODEBUGMSG=" END OF VALIDATETARGETACCOUNT - "+
-              FORMAT("NAMEFOUND:9",NAMEFOUND)+
-              FORMAT("   SLFOUND:9",SLFOUND)
+ BNODEBUGMSG=" END OF VALIDATETARGETACCOUNT - "+FORMAT("VALIDATESTATUS:9",VALIDATESTATUS)
  CALL BNODEBUGMSGADD
 
- IF TARGETACCOUNT<>THISACCOUNT AND
-    NAMEFOUND=TRUE AND
-    SLFOUND=TRUE THEN
-  TARGETACCOUNTVALIDATION=TRUE
 END [PROCEDURE]
 
+PROCEDURE FINDBYSHARECODE
+
+ SLFOUND=FALSE
+ FOR EACH SHARE WITH (SHARE:CLOSEDATE=DATENULL AND
```

Comment:
That probably does make a little more sense.  But I think I could even argue for removing it all together since maybe the ALLOWABLESHARECODES should only apply to when an ID is actually entered -- and not in the case where we have the logic to find a share because no ID was entered.  Thoughts??

Reply:
Made a change.